Assets/Scripts/Models/Character.cs

```diff
@@ -591,11 +591,14 @@ private void GetNewJob()
             }
         }
 
-        if (needPercent > 50 && needPercent < 100 && need.RestoreNeedFurn != null)
+        if (needPercent > 50 && needPercent <= 100 && need.RestoreNeedFurn != null)
         {
-            if (World.Current.CountFurnitureType(need.RestoreNeedFurn.ObjectType) > 0)
+            IEnumerable<Furniture> furnitures = World.Current.furnitures.Where(furniture => furniture.ObjectType == need.RestoreNeedFurn.ObjectType && furniture.IsUsable() == true);
```

Comment:
`IEnumerable` is just a query. It is executed each time you try to iterate with it.
Usually it is bad when you have to iterate through `IEnumerable` several times. This is because list of object could change between iterations and this would lead to situation when you iterate with two different lists.
If you need to use `IEnumerable` several time it is good idea to execute by `.ToList()` `.ToArray()` call